### ISSUE TITLE
Support for Date values in Prisma.JsonValue

### ISSUE THREAD

**Issue by tamirvertesh** [prisma/prisma#18552]
## Problem

I need to define a field in my schema (for mongodb) that is a dictionary between a string and a date (the keys are unknown at build time).
The closest way I found to doing this, is with a `Json` type. However, Prisma does not support `Date` values, but only number, string, etc.

## Suggested solution

Add support for a Date (MongoDB Timestamp) in Prisma.JsonValue.

**Comment by janpio** [comment]
Does Json have a Date type? Not according to e.g. https://www.w3schools.com/js/js_json_datatypes.asp

**Comment by jkomyno** [comment]
@janpio that's true, but dates could be serialized into ISO 8601 strings (e.g., in the `2023-07-15` format), and timestamps / timestampstz could adopt similar serialization practices. I don't think this is supported out of the box in mongodb, though
